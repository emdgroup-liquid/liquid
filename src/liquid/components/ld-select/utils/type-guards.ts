export const isLdOption = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptionElement | HTMLLdOptionInternalElement =>
  ["LD-OPTION", "LD-OPTION-INTERNAL"].includes((el as HTMLElement)?.tagName);

export const isLdOptgroup = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptgroupElement | HTMLLdOptgroupInternalElement =>
  ["LD-OPTGROUP", "LD-OPTGROUP-INTERNAL"].includes(
    (el as HTMLElement)?.tagName,
  );

export const isLdOptionInternal = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptionInternalElement =>
  ["LD-OPTION-INTERNAL"].includes((el as HTMLElement)?.tagName);

export const isLdOptgroupInternal = (
  el: HTMLElement | Node | EventTarget,
): el is HTMLLdOptgroupInternalElement =>
  ["LD-OPTGROUP-INTERNAL"].includes((el as HTMLElement)?.tagName);

type HTMLLdOptInternal =
  | HTMLLdOptionInternalElement
  | HTMLLdOptgroupInternalElement;
export const isLdOptInternalHidden = (
  opt: HTMLLdOptInternal,
): opt is
  | (HTMLLdOptInternal & {
      hidden: true;
    })
  | (HTMLLdOptInternal & {
      filtered: true;
    }) => {
  return opt.hidden || opt.filtered;
};
