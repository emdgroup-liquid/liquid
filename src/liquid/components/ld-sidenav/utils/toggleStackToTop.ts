import { LdSidenavNavitem } from '../ld-sidenav-navitem/ld-sidenav-navitem'

export const toggleStackToTop = (el: HTMLElement, stacked: boolean) => {
  const outerSlider = el.closest('ld-sidenav-slider')
  if (!outerSlider) return

  const sidenavPaddingY =
    parseInt(
      window
        .getComputedStyle(outerSlider)
        .getPropertyValue('--ld-sidenav-padding-y')
    ) * 16

  // If not stacked, put everything back in place.
  if (!stacked) {
    el.style.removeProperty('--ld-slider-navitem-move-up')
    return
  }

  // Else, calculate how much vertical space is taken by stacked items above
  // and move the nav item up accordingly.
  let totalSpaceAbove = 0
  let totalSpaceOccupiedAbove = 0
  for (const elem of Array.from(el.parentElement.children)) {
    if (el === elem) break

    if (
      elem.tagName === 'LD-SIDENAV-SEPARATOR' ||
      (elem.tagName === 'LD-SIDENAV-NAVITEM' &&
        !['secondary', 'tertiary'].includes(
          (elem as unknown as LdSidenavNavitem).mode
        ))
    ) {
      totalSpaceOccupiedAbove += elem.getBoundingClientRect().height
    } else {
      totalSpaceAbove += sidenavPaddingY
    }
    totalSpaceAbove += elem.getBoundingClientRect().height
  }

  const spaceToMoveUp = totalSpaceAbove - totalSpaceOccupiedAbove
  el.style.setProperty('--ld-slider-navitem-move-up', `-${spaceToMoveUp}px`)
}
