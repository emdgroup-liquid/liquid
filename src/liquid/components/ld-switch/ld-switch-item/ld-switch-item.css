:host,
.ld-switch-item {
  display: inline-flex;
  position: relative;
  line-height: 1.2;

  label {
    display: flex;
    align-items: center;
    border: 0;
    border-radius: 0;
    background-clip: padding-box;
  }

  .ld-switch-item__faux {
    height: var(--ld-switch-item-height, 2.5rem);
    font-family: var(--ld-switch-item-font, var(--ld-font-body));
    font-size: var(--ld-switch-item-font-size, 1rem);
    padding-left: var(--ld-switch-item-padding-y, 0.75rem);
    padding-right: var(--ld-switch-item-padding-y, 0.75rem);
    display: flex;
    align-items: center;
    justify-content: center;

    background-color: var(
      --ld-switch-item-bg-col,
      var(--ld-thm-primary-alpha-low)
    );
    color: var(--ld-switch-item-text-col, var(--ld-thm-primary));
  }

  input {
    width: 1px;
    height: 1px;
    border: 0;
    clip: rect(0 0 0 0);
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    appearance: none;

    &:checked {
      ~ .ld-switch-item__faux {
        background-color: var(
          --ld-switch-item-bg-col-selected,
          var(--ld-thm-primary)
        );
        color: var(--ld-switch-item-text-col-selected, var(--ld-col-wht));
      }
    }

    &:where(:disabled),
    &:where([aria-disabled='true']) {
      ~ .ld-switch-item__faux {
        opacity: 0.2;
      }
    }

    &:where(:not(:disabled):not([aria-disabled='true']):not([readonly])) {
      ~ .ld-switch-item__faux {
        cursor: pointer;

        @media (hover: hover) {
          &:hover {
            background-color: var(
              --ld-switch-item-bg-col-hover,
              var(--ld-thm-primary-hover)
            );
            color: var(--ld-switch-item-text-col-active, var(--ld-col-wht));
          }
        }
      }

      &:focus:focus-visible {
        ~ .ld-switch-item__faux {
          background-color: var(
            --ld-switch-item-bg-col-focus,
            var(--ld-thm-primary-focus)
          );
          color: var(--ld-switch-item-text-col-active, var(--ld-col-wht));
        }
      }

      &:active,
      &:active:focus-visible {
        ~ .ld-switch-item__faux {
          background-color: var(
            --ld-switch-item-bg-col-active,
            var(--ld-thm-primary-active)
          );
          color: var(--ld-switch-item-text-col-active, var(--ld-col-wht));
        }
      }
    }
  }
}
