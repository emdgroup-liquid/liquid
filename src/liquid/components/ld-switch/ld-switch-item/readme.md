---
eleventyNavigation:
  key: Switch Item
  parent: Switch
layout: layout.njk
title: Switch Item
permalink: components/ld-switch/ld-switch-item/
---

# ld-switch-item

The `ld-switch-item` component is a subcomponent for `ld-switch`.

Please refer to the [`ld-switch` documentation](components/ld-switch) for usage examples.


<!-- Auto Generated Below -->


## Properties

| Property             | Attribute     | Description                                                            | Type      | Default     |
| -------------------- | ------------- | ---------------------------------------------------------------------- | --------- | ----------- |
| `autofocus`          | `autofocus`   | Automatically focus the form control when the page is loaded.          | `boolean` | `false`     |
| `checked`            | `checked`     | Indicates whether the radio button is selected.                        | `boolean` | `false`     |
| `disabled`           | `disabled`    | Disabled state of the radio.                                           | `boolean` | `undefined` |
| `form`               | `form`        | Associates the control with a form element.                            | `string`  | `undefined` |
| `label` _(required)_ | `label`       | The label of the switch item                                           | `string`  | `undefined` |
| `ldTabindex`         | `ld-tabindex` | Tab index of the input.                                                | `number`  | `undefined` |
| `name`               | `name`        | Used to specify the name of the control.                               | `string`  | `undefined` |
| `readonly`           | `readonly`    | The value is not editable.                                             | `boolean` | `undefined` |
| `required`           | `required`    | Set this property to `true` in order to mark the checkbox as required. | `boolean` | `undefined` |
| `value` _(required)_ | `value`       | The input value.                                                       | `string`  | `undefined` |


## Events

| Event                | Description                                                       | Type                   |
| -------------------- | ----------------------------------------------------------------- | ---------------------- |
| `ldswitchitemchange` | Emitted when the input value changed and the element loses focus. | `CustomEvent<string>`  |
| `ldswitchiteminput`  | Emitted when the input value changed.                             | `CustomEvent<boolean>` |


## Methods

### `focusInner() => Promise<void>`

Sets focus on the radio button.

#### Returns

Type: `Promise<void>`




## Shadow Parts

| Part     | Description |
| -------- | ----------- |
| `"root"` |             |


----------------------------------------------

 
