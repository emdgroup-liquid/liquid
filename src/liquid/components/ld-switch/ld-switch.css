:host,
.ld-switch {
  /* layout */
  --ld-switch-max-height: 2.5rem;

  /* colors */
  --ld-switch-text-col: var(--ld-thm-primary);
  --ld-switch-bg-col: var(--ld-thm-primary-alpha-low);

  --ld-switch-bg-col-hover: var(--ld-thm-primary-hover);
  --ld-switch-bg-col-active: var(--ld-thm-primary-active);
  --ld-switch-bg-col-focus: var(--ld-thm-primary-focus);

  --ld-switch-selected-text-col: var(--ld-col-wht);
  --ld-switch-selected-bg-col: var(--ld-thm-primary);

  /* ld-switch-item vars */
  --ld-switch-item-padding-y: 0.75rem;
  --ld-switch-item-font-size: 1rem;

  position: relative;
  display: block;

  fieldset {
    position: relative;
    min-width: 0;
    border: 0;
  }

  legend {
    position: absolute;
    margin: -1px;
    border: 0;
    padding: 0;
    width: 1px;
    height: 1px;
    overflow: hidden;
    clip: rect(0 0 0 0);
    text-transform: none;
    white-space: nowrap;
  }

  .ld-switch_item-container {
    position: relative;
    display: inline-flex;
    border-radius: 0.25rem;
    overflow: hidden;
  }
}

:host(.ld-switch--sm),
.ld-switch--sm {
  --ld-switch-item-padding-y: 0.625rem;
  --ld-switch-item-font-size: 0.875rem;
}

:host(.ld-switch--lg),
.ld-switch--lg {
  --ld-switch-item-padding-y: 1rem;
  --ld-switch-item-font-size: 1.125rem;
}

:host(.ld-switch--brand-color),
.ld-switch--brand-color {
  --ld-switch-item-text-col: var(--ld-thm-primary-highlight);
  --ld-switch-item-bg-col: rgba(255, 255, 255, 0.2);

  --ld-switch-item-bg-col-hover: var(--ld-col-wht-alpha-high);
  --ld-switch-item-bg-col-active: var(--ld-col-wht-alpha-medium);
  --ld-switch-item-bg-col-focus: var(--ld-col-wht-alpha-high);
  --ld-switch-item-bg-col-selected: var(--ld-col-wht);

  --ld-switch-item-text-col-selected: var(--ld-thm-primary);
  --ld-switch-item-text-col-active: var(--ld-switch-item-text-col-selected);
}
