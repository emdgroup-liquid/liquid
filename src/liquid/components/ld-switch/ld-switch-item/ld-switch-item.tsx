import {
  Component,
  Element,
  Event,
  EventEmitter,
  Host,
  Method,
  h,
  Prop,
  State,
  Watch,
} from '@stencil/core'
import { cloneAttributes } from '../../../utils/cloneAttributes'

@Component({
  tag: 'ld-switch-item',
  styleUrl: 'ld-switch-item.css',
  shadow: true,
})
export class LdSwitchItem implements InnerFocusable, ClonesAttributes {
  @Element() el: HTMLInputElement

  private attributesObserver: MutationObserver

  private input: HTMLInputElement
  private hiddenInput: HTMLInputElement

  /** Automatically focus the form control when the page is loaded. */
  @Prop() autofocus = false

  /** Indicates whether the radio button is selected. */
  @Prop({ mutable: true }) checked = false

  /** Disabled state of the radio. */
  @Prop() disabled: boolean

  /** Associates the control with a form element. */
  @Prop() form?: string

  /** Tab index of the input. */
  @Prop() ldTabindex: number | undefined

  /** Used to specify the name of the control. */
  @Prop() name: string

  /** The value is not editable. */
  @Prop() readonly?: boolean

  /** Set this property to `true` in order to mark the checkbox as required. */
  @Prop() required: boolean

  /** The input value. */
  @Prop() value!: string

  /** The label of the switch item */
  @Prop() label!: string

  @State() clonedAttributes

  /** Emitted when the input value changed and the element loses focus. */
  @Event() ldswitchitemchange: EventEmitter<string>

  /** Emitted when the input value changed. */
  @Event() ldswitchiteminput: EventEmitter<boolean>

  /** Sets focus on the radio button. */
  @Method()
  async focusInner() {
    if (this.input !== undefined) {
      this.input.focus()
    }
  }

  @Watch('disabled')
  @Watch('checked')
  @Watch('name')
  @Watch('value')
  updateHiddenInput() {
    const outerForm = this.el.closest('form')
    if (!this.hiddenInput && this.name && (outerForm || this.form)) {
      this.createHiddenInput()
    }

    if (this.hiddenInput) {
      if (!this.name) {
        this.hiddenInput.remove()
        this.hiddenInput = undefined
        return
      }

      this.hiddenInput.name = this.name
      this.hiddenInput.checked = this.checked

      if (this.value) {
        this.hiddenInput.value = this.value
      } else {
        this.hiddenInput.removeAttribute('value')
      }

      if (this.form) {
        this.hiddenInput.setAttribute('form', this.form)
      } else if (this.hiddenInput.getAttribute('form')) {
        if (outerForm) {
          this.hiddenInput.removeAttribute('form')
        } else {
          this.hiddenInput.remove()
          this.hiddenInput = undefined
        }
      }
    }
  }

  private createHiddenInput() {
    this.hiddenInput = document.createElement('input')
    this.hiddenInput.type = 'radio'
    this.hiddenInput.style.visibility = 'hidden'
    this.hiddenInput.style.position = 'absolute'
    this.hiddenInput.style.pointerEvents = 'none'
    this.el.appendChild(this.hiddenInput)
  }

  private handleKeyDown = (ev: KeyboardEvent) => {
    switch (ev.key) {
      case 'ArrowUp':
      case 'ArrowLeft': {
        ev.preventDefault()
        this.focusAndSelect('prev')
        return
      }
      case 'ArrowDown':
      case 'ArrowRight': {
        ev.preventDefault()
        this.focusAndSelect('next')
        return
      }
    }
  }

  private handleChange = (event: InputEvent) => {
    this.el.dispatchEvent(new InputEvent('change', event))
    this.ldswitchitemchange.emit(this.el.value)
  }

  private handleClick = (ev?: MouseEvent) => {
    if (
      this.disabled ||
      this.el.getAttribute('aria-disabled') === 'true' ||
      this.readonly
    ) {
      ev?.preventDefault()
      return
    }

    if (this.checked) return

    // Uncheck radios with same name.
    if (this.name) {
      // Attribute selector fails in test env, hance filtering with js below.
      Array.from(document.querySelectorAll('ld-switch-item'))
        .filter(
          (ldSwitchItem) => ldSwitchItem.getAttribute('name') === this.name
        )
        .forEach((ldSwitchItem) => {
          ldSwitchItem.checked = false
        })
    }

    this.checked = true

    if (!ev.isTrusted) {
      // This happens, when a click event is dispatched on the host element
      // from the outside i.e. on click on a parent ld-label element.
      this.el.dispatchEvent(
        new InputEvent('input', { bubbles: true, composed: true })
      )
      this.handleInput()
      this.el.dispatchEvent(new InputEvent('change', { bubbles: true }))
      this.ldswitchitemchange.emit(this.el.value)
    }
  }

  private handleInput = () => {
    this.ldswitchiteminput.emit(this.checked)
  }

  private focusAndSelect(dir: 'next' | 'prev') {
    const ldSwitchItems = Array.from(
      document.querySelectorAll('ld-switch-item')
    ).filter((ldSwitchItem) => ldSwitchItem.getAttribute('name') === this.name)

    ldSwitchItems.forEach((ldSwitchItem, index) => {
      if (ldSwitchItem === (this.el as unknown as HTMLLdSwitchItemElement)) {
        const targetLdSwitchItem =
          ldSwitchItems[index + (dir === 'next' ? 1 : -1)]
        if (targetLdSwitchItem) {
          targetLdSwitchItem.focusInner()
          targetLdSwitchItem.click()
        }
      }
    })
  }

  @Watch('label')
  componentWillLoad() {
    this.attributesObserver = cloneAttributes.call(this, ['label'])

    const outerForm = this.el.closest('form')

    if (this.name && (outerForm || this.form)) {
      this.createHiddenInput()
      this.hiddenInput.checked = this.checked
      this.hiddenInput.name = this.name

      if (this.form) {
        this.hiddenInput.setAttribute('form', this.form)
      }

      if (this.value) {
        this.hiddenInput.value = this.value
      }
    }
  }

  componentDidLoad() {
    if (this.autofocus) {
      this.focusInner()
    }
  }

  disconnectedCallback() {
    this.attributesObserver?.disconnect()
  }

  render() {
    return (
      <Host part="root" class="ld-switch-item" onClick={this.handleClick}>
        <label>
          <input
            type="radio"
            {...this.clonedAttributes}
            part="input focusable"
            onChange={this.handleChange}
            onInput={this.handleInput}
            onKeyDown={this.handleKeyDown}
            ref={(ref) => (this.input = ref)}
            disabled={this.disabled}
            checked={this.checked}
            tabIndex={this.checked ? this.ldTabindex : -1}
          />
          <span class="ld-switch-item__faux">{this.label}</span>
        </label>
      </Host>
    )
  }
}
