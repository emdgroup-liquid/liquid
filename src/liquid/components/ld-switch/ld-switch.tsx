import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  Host,
  Prop,
  Watch,
} from '@stencil/core'
import { getClassNames } from '../../utils/getClassNames'

/**
 * @virtualProp ref - reference to component
 * @virtualProp {string | number} key - for tracking the node's identity when working with lists
 * @part switch_item-container - Container wrapping the slot
 */

@Component({
  tag: 'ld-switch',
  styleUrl: 'ld-switch.css',
  shadow: true,
})
export class LdSwitch {
  @Element() el: HTMLElement

  /** Size of the switch. */
  @Prop() size?: 'sm' | 'md' | 'lg'

  /** Defines switch custom color */
  @Prop() brandColor?: boolean

  /** Defines a description of the contents of the switch component.*/
  @Prop() label!: string

  /** Automatically focus the form control when the page is loaded. */
  @Prop() autofocus = false

  /** Disabled state of the switch. */
  @Prop() disabled: boolean

  /** Alternative disabled state that keeps element focusable */
  @Prop() ariaDisabled: string

  /** Associates the control with a form element. */
  @Prop() form?: string

  /** Used to specify the name of the control. */
  @Prop() name!: string

  /** The value is not editable. */
  @Prop() readonly?: boolean

  /** Set this property to `true` in order to mark the switch as required. */
  @Prop() required: boolean

  /**
   * Emitted with the value of the selected switch item.
   */
  @Event() ldswitchchange: EventEmitter<string>

  private handleLdSwitchItemChange = (ev: CustomEvent<undefined>) => {
    ev.stopImmediatePropagation()
    const currentLdSwitchItem = ev.target as HTMLLdSwitchItemElement
    this.ldswitchchange.emit(currentLdSwitchItem.value)
  }

  @Watch('name')
  @Watch('disabled')
  componentWillLoad() {
    const ldSwitchItems = this.el.querySelectorAll('ld-switch-item')
    const hasAnyChecked =
      ldSwitchItems.length > 0 &&
      Array.from(ldSwitchItems).some((ldSwitchItem) => ldSwitchItem.checked)

    ldSwitchItems.forEach((ldSwitchItem) => {
      if (this.name) {
        ldSwitchItem.setAttribute('name', this.name)
      }

      if (this.readonly) {
        ldSwitchItem.setAttribute('readonly', this.readonly.toString())
      }

      if (this.disabled) {
        ldSwitchItem.disabled = this.disabled
      }

      if (this.required) {
        ldSwitchItem.required = this.required
      }
    })

    if (ldSwitchItems.length > 0 && !hasAnyChecked) {
      ldSwitchItems[0].checked = true
    }
  }

  render() {
    return (
      <Host
        class={getClassNames([
          'ld-switch',
          this.size && `ld-switch--${this.size}`,
          this.brandColor && `ld-switch--brand-color`,
        ])}
        onLdswitchitemchange={this.handleLdSwitchItemChange}
      >
        <fieldset part="fieldset">
          <legend>{this.label}</legend>
          <div class="ld-switch_item-container" part="switch_item-container">
            <slot></slot>
          </div>
        </fieldset>
      </Host>
    )
  }
}
