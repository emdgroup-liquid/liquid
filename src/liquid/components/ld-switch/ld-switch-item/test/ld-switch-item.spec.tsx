import { newSpecPage } from '@stencil/core/testing'
import { LdSwitch } from '../../ld-switch'
import { LdSwitchItem } from '../ld-switch-item'
import '../../../../utils/mutationObserver'

const components = [LdSwitch, LdSwitchItem]

describe('ld-switch-item', () => {
  describe('modifiers', () => {
    it('size', async () => {
      const page = await newSpecPage({
        components,
        html: `
        <ld-switch label="Food Type" name="foodType" size="lg">
          <ld-switch-item label="Fruits" value="fruits"></ld-switch-item>
          <ld-switch-item label="Vegetables" value="vegetables"></ld-switch-item>
          <ld-switch-item label="Nuts" value="nuts"></ld-switch-item>
        </ld-switch>
        `,
      })

      expect(page.root).toMatchSnapshot()
    })

    it('disabled', async () => {
      const page = await newSpecPage({
        components,
        html: `
        <ld-switch label="Food Type" name="foodType" disabled>
          <ld-switch-item label="Fruits" value="fruits"></ld-switch-item>
          <ld-switch-item label="Vegetables" value="vegetables"></ld-switch-item>
          <ld-switch-item label="Nuts" value="nuts"></ld-switch-item>
        </ld-switch>
        `,
      })

      expect(page.root).toMatchSnapshot()
    })

    it('single disabled', async () => {
      const page = await newSpecPage({
        components,
        html: `
        <ld-switch label="Food Type" name="foodType">
          <ld-switch-item label="Fruits" value="fruits"></ld-switch-item>
          <ld-switch-item label="Vegetables" value="vegetables" disabled></ld-switch-item>
          <ld-switch-item label="Nuts" value="nuts"></ld-switch-item>
        </ld-switch>
        `,
      })

      expect(page.root).toMatchSnapshot()
    })

    it('brand-color', async () => {
      const page = await newSpecPage({
        components,
        html: `
        <ld-switch label="Food Type" name="foodType" brand-color>
          <ld-switch-item label="Fruits" value="fruits"></ld-switch-item>
          <ld-switch-item label="Vegetables" value="vegetables" disabled></ld-switch-item>
          <ld-switch-item label="Nuts" value="nuts"></ld-switch-item>
        </ld-switch>
        `,
      })

      expect(page.root).toMatchSnapshot()
    })
  })
})
