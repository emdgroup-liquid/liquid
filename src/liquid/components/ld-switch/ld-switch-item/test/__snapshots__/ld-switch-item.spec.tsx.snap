// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ld-switch-item modifiers brand-color 1`] = `
<ld-switch brand-color="" class="ld-switch ld-switch--brand-color" label="Food Type" name="foodType">
  <mock:shadow-root>
    <fieldset part="fieldset">
      <legend>
        Food Type
      </legend>
      <div class="ld-switch_item-container" part="switch_item-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ld-switch-item class="ld-switch-item" label="Fruits" name="foodType" part="root" value="fruits">
    <mock:shadow-root>
      <label>
        <input checked="" name="foodType" part="input focusable" type="radio" value="fruits">
        <span class="ld-switch-item__faux">
          Fruits
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" disabled="" label="Vegetables" name="foodType" part="root" value="vegetables">
    <mock:shadow-root>
      <label>
        <input disabled="" name="foodType" part="input focusable" tabindex="-1" type="radio" value="vegetables">
        <span class="ld-switch-item__faux">
          Vegetables
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" label="Nuts" name="foodType" part="root" value="nuts">
    <mock:shadow-root>
      <label>
        <input name="foodType" part="input focusable" tabindex="-1" type="radio" value="nuts">
        <span class="ld-switch-item__faux">
          Nuts
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
</ld-switch>
`;

exports[`ld-switch-item modifiers disabled 1`] = `
<ld-switch class="ld-switch" disabled="" label="Food Type" name="foodType">
  <mock:shadow-root>
    <fieldset part="fieldset">
      <legend>
        Food Type
      </legend>
      <div class="ld-switch_item-container" part="switch_item-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ld-switch-item class="ld-switch-item" label="Fruits" name="foodType" part="root" value="fruits">
    <mock:shadow-root>
      <label>
        <input checked="" disabled="" name="foodType" part="input focusable" type="radio" value="fruits">
        <span class="ld-switch-item__faux">
          Fruits
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" label="Vegetables" name="foodType" part="root" value="vegetables">
    <mock:shadow-root>
      <label>
        <input disabled="" name="foodType" part="input focusable" tabindex="-1" type="radio" value="vegetables">
        <span class="ld-switch-item__faux">
          Vegetables
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" label="Nuts" name="foodType" part="root" value="nuts">
    <mock:shadow-root>
      <label>
        <input disabled="" name="foodType" part="input focusable" tabindex="-1" type="radio" value="nuts">
        <span class="ld-switch-item__faux">
          Nuts
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
</ld-switch>
`;

exports[`ld-switch-item modifiers single disabled 1`] = `
<ld-switch class="ld-switch" label="Food Type" name="foodType">
  <mock:shadow-root>
    <fieldset part="fieldset">
      <legend>
        Food Type
      </legend>
      <div class="ld-switch_item-container" part="switch_item-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ld-switch-item class="ld-switch-item" label="Fruits" name="foodType" part="root" value="fruits">
    <mock:shadow-root>
      <label>
        <input checked="" name="foodType" part="input focusable" type="radio" value="fruits">
        <span class="ld-switch-item__faux">
          Fruits
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" disabled="" label="Vegetables" name="foodType" part="root" value="vegetables">
    <mock:shadow-root>
      <label>
        <input disabled="" name="foodType" part="input focusable" tabindex="-1" type="radio" value="vegetables">
        <span class="ld-switch-item__faux">
          Vegetables
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" label="Nuts" name="foodType" part="root" value="nuts">
    <mock:shadow-root>
      <label>
        <input name="foodType" part="input focusable" tabindex="-1" type="radio" value="nuts">
        <span class="ld-switch-item__faux">
          Nuts
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
</ld-switch>
`;

exports[`ld-switch-item modifiers size 1`] = `
<ld-switch class="ld-switch ld-switch--lg" label="Food Type" name="foodType" size="lg">
  <mock:shadow-root>
    <fieldset part="fieldset">
      <legend>
        Food Type
      </legend>
      <div class="ld-switch_item-container" part="switch_item-container">
        <slot></slot>
      </div>
    </fieldset>
  </mock:shadow-root>
  <ld-switch-item class="ld-switch-item" label="Fruits" name="foodType" part="root" value="fruits">
    <mock:shadow-root>
      <label>
        <input checked="" name="foodType" part="input focusable" type="radio" value="fruits">
        <span class="ld-switch-item__faux">
          Fruits
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" label="Vegetables" name="foodType" part="root" value="vegetables">
    <mock:shadow-root>
      <label>
        <input name="foodType" part="input focusable" tabindex="-1" type="radio" value="vegetables">
        <span class="ld-switch-item__faux">
          Vegetables
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
  <ld-switch-item class="ld-switch-item" label="Nuts" name="foodType" part="root" value="nuts">
    <mock:shadow-root>
      <label>
        <input name="foodType" part="input focusable" tabindex="-1" type="radio" value="nuts">
        <span class="ld-switch-item__faux">
          Nuts
        </span>
      </label>
    </mock:shadow-root>
  </ld-switch-item>
</ld-switch>
`;
